freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/backup/cloud/AdlsGen2BackupConfigGenerator.java

```diff
@@ -0,0 +1,63 @@
+package com.sequenceiq.freeipa.service.freeipa.backup.cloud;
+
+import java.nio.file.Paths;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.common.exception.CloudbreakServiceException;
+import com.sequenceiq.cloudbreak.telemetry.fluent.cloud.AdlsGen2Config;
+
+@Component
+public class AdlsGen2BackupConfigGenerator extends CloudBackupConfigGenerator<AdlsGen2Config> {
```

Comment:
why duplicate this kind of classes/implementations from common ? most of the code looks duplication

Reply:
Its not an outright copy.  Right now it is using very similar logic on the parsing of the adls container and s3 buckets paths, but the functions are slightly different with the base backups folder being different and a changed call path. 

As well, later on we will be moving to a dedicated selection of the backup storage location to a stand-alone bucket/container but the UI and flow work was unable to be done in the time needed for that call.  This allows for the seperate parsing mechanism for the backup once it moves away from the telementry storage location.

Reply:
you can make things more abstract, but fine...btw it's nothing to do with Fluent but it has many references for it, so that looks logically invalid for me (for example: using fluent cluster type that is used as a string in fluentd related configs - so if i would change that for logging it would affect this code as well)

Reply:
Fair enough.  I refactored it to be more independent.  There is a single value enum because It will be added to later on when it gets more options.